src/asset-swapper/utils/market_operation_utils/orders.ts

```diff
@@ -587,6 +636,101 @@ export function createNativeOptimizedOrder(
+    fill: Fill<NativeFillData>,
+    multiHop: NativeOrderWithFillableAmounts[],
+    side: MarketOperation,
+) : | OptimizedMarketOrderBase<NativeFillData>[] {
+    const fillData = fill.fillData;
+    //const [makerAmount, takerAmount] = getFillTokenAmounts(fill, side);
+    // const [makerAmount1, takerAmount1] = getMultiHopFillTokenAmounts(multiHop[0], side);
+    // const [makerAmount2, takerAmount2] = getMultiHopFillTokenAmounts(multiHop[1], side);
+    
+    const base = {
+        type: fill.type,
+        source: ERC20BridgeSource.Native,
+        makerToken: fillData.order.makerToken,
+        takerToken: fillData.order.takerToken,
+        fill: cleanFillForExport(fill),
+    };
+
+    // export interface Fill<TFillData extends FillData = FillData> {
+    //     [x: string]: any;
+    //     // basic data for every fill
+    //     source: ERC20BridgeSource;
+    //     // TODO jacob people seem to agree  that orderType here is more readable
+    //     type: FillQuoteTransformerOrderType; // should correspond with TFillData
+    //     fillData: TFillData;
+    //     // Unique ID of the original source path this fill belongs to.
+    //     // This is generated when the path is generated and is useful to distinguish
+    //     // paths that have the same `source` IDs but are distinct (e.g., Curves).
+    //     sourcePathId: string;
+    //     // See `SOURCE_FLAGS`.
+    //     flags: bigint;
+    //     // Input fill amount (taker asset amount in a sell, maker asset amount in a buy).
+    //     input: BigNumber;
+    //     // Output fill amount (maker asset amount in a sell, taker asset amount in a buy).
+    //     output: BigNumber;
+    //     // The output fill amount, adjusted by fees.
+    //     adjustedOutput: BigNumber;
+    //     // The expected gas cost of this fill
+    //     gas: number;
+    // }
+
+    let optimizedOtcOrders: OptimizedMarketOrderBase<NativeFillData>[] = [];
+    //_.omit(fill, ['flags', 'fillData', 'sourcePathId', 'source', 'type']) as Fill;
+    let firstOrderExpiry: BigNumber, firstOrderNonce: BigNumber, firstOrderNonceBucket: BigNumber, firstOrderMaker: string;
+    multiHop.forEach( (order, index) => {
+        let optimizedOtcOrder: OptimizedMarketOrderBase<NativeFillData>;
+        let otcFillData: NativeOtcOrderFillData;
+        let otcFill: Omit<Fill<FillData>, 'flags' | 'fillData'| 'sourcePathId'| 'source'| 'type'>;
+        order.order = (order.order as OtcOrderFields);
+        let { expiry, nonce, nonceBucket } = OtcOrder.parseExpiryAndNonce(order.order.expiryAndNonce);
+        let maker = order.order.maker;
+        if(index == 0){
+            firstOrderExpiry = expiry;
+            firstOrderNonce = nonce;
+            firstOrderNonceBucket = nonceBucket;
+            firstOrderMaker = order.order.maker;
+        }
+        else {
+            //if we are trading with the same market maker, increment the nonce on the second otc order
```

Comment:
could we potentially have a nonce collision with other otc orders if they happen to be filled in the same block?